core/modules/views/templates/views.theme.inc

```diff
@@ -1138,6 +1138,11 @@ function theme_views_form_views_form($variables) {
 function theme_views_mini_pager($variables) {
   global $pager_page_array, $pager_total;
 
+  // Pager is only needed if there are results.
```

Comment:
FWIW core pager bails out with this code:
```
  if (!isset($pager_page_array[$element]) || empty($pager_total)) {
    return '';
  }
```
Dont know if this could be an issue, but I'd say copying what core does likely safer?

Reply:
Yes, make them the same.

Reply:
Done.